### PR TITLE
Fixed #26510 -- Allowed dim/trim/precision as WKTWriter init arguments

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: tests/gis_tests/geos_tests/test_io.py [hunk 1]
```diff
@@ -101,9 +101,8 @@ def test04_wkbwriter(self):
 
         # Ensuring bad output dimensions are not accepted
         for bad_outdim in (-1, 0, 1, 4, 423, 'foo', None):
-            # Equivalent of `wkb_w.outdim = bad_outdim`
             with self.assertRaises(ValueError):
```
Comment (timgraham) [inline tests/gis_tests/geos_tests/test_io.py]: Could update this to use `assertRaisesMessage` while you're here.

File: docs/ref/contrib/gis/geos.txt [hunk 2]
```diff
@@ -1047,7 +1049,11 @@ geometry should be included with the WKB representation.  Example::
     >>> wkb_w.write_hex(pnt)
     '0101000020E6100000000000000000F03F000000000000F03F'
 
-.. class:: WKTWriter
+.. class:: WKTWriter(dim=2, trim=False, precision=None)
+
+    This class allows outputting the WKT representation of a geometry. See the
+    :attr:`WKBWriter.outdim`, :attr:`trim` and :attr:`precision` attributes for
```
Comment (timgraham) [inline docs/ref/contrib/gis/geos.txt]: , and

**Comment by timgraham** [comment]
Any versionadded/changed or release notes needed?

**Comment by claudep** [comment]
I'll add a versionchanged, however I wouldn't load the release notes for mostly internally used stuff.

**Pull request closed without merging**